src/App.js

```diff
@@ -1,27 +1,22 @@
 import './App.css';
 import Post from './post';
+import { renderAllPosts } from './api-service';
 import { useState, useEffect } from 'react';
 import { NavLink } from 'react-router-dom';
 import { useNavigate } from 'react-router-dom';
 
-function App({ auth, setUserId, userToken }) {
+function App({ setUserId, userToken }) {
   const [allPosts, setAllPosts] = useState([]);
   const navigate = useNavigate();
 
-  console.log(userToken, 'userToken в App');
+  // console.log(renderAllPosts, 'renderAllPosts');
 
   function navigateToAddPost() {
     navigate('/add-post');
   }
 
-  function renderAllPosts() {
-    fetch('https://webdev-hw-api.vercel.app/api/v1/prod/instapro')
-      .then((response) => response.json())
-      .then((data) => setAllPosts(data.posts))
-  }
-
   useEffect(() => {
-    renderAllPosts()
+    renderAllPosts(setAllPosts)
```

Comment:
давай сделаем так, чтобы setAllPost вызывался здесь, в компоненте. Это же логика компонента, пусть она тут и будет, а в апи сервисе будет только логика связанная с запросом в апи и обработкой ответа

Так будет правильнее с точки зрения data flow в приложении:
```suggestion
   renderAllPosts().then(posts => {
      setAllPosts(posts)
   });
   
```

Reply:
Разберись с этим вызовом апи, когда поймешь принцип работы с промисами, то остальные тебе сделать будет легко